substrate/frame/bounties/src/benchmarking.rs

```diff
@@ -17,87 +17,163 @@
+impl<AssetKind> ArgumentsFactory<AssetKind> for ()
+where
+	AssetKind: FromEntropy,
+{
+	fn create_asset_kind(seed: u32) -> AssetKind {
+		AssetKind::from_entropy(&mut seed.encode().as_slice()).unwrap()
 	}
 }
 
-// Create bounties that are approved for use in `on_initialize`.
-fn create_approved_bounties<T: Config<I>, I: 'static>(n: u32) -> Result<(), BenchmarkError> {
-	for i in 0..n {
-		let (caller, _curator, _fee, value, reason) =
-			setup_bounty::<T, I>(i, T::MaximumReasonLength::get());
-		Bounties::<T, I>::propose_bounty(RawOrigin::Signed(caller).into(), value, reason)?;
-		let bounty_id = BountyCount::<T, I>::get() - 1;
-		let approve_origin =
-			T::SpendOrigin::try_successful_origin().map_err(|_| BenchmarkError::Weightless)?;
-		Bounties::<T, I>::approve_bounty(approve_origin, bounty_id)?;
-	}
-	ensure!(BountyApprovals::<T, I>::get().len() == n as usize, "Not all bounty approved");
-	Ok(())
+const SEED: u32 = 0;
+
+fn set_block_number<T: Config<I>, I: 'static>(n: BlockNumberFor<T, I>) {
+	<T as pallet_treasury::Config<I>>::BlockNumberProvider::set_block_number(n);
 }
 
 // Create the pre-requisite information needed to create a treasury `propose_bounty`.
 fn setup_bounty<T: Config<I>, I: 'static>(
-	u: u32,
-	d: u32,
-) -> (T::AccountId, T::AccountId, BalanceOf<T, I>, BalanceOf<T, I>, Vec<u8>) {
-	let caller = account("caller", u, SEED);
-	let value: BalanceOf<T, I> = T::BountyValueMinimum::get().saturating_mul(100u32.into());
-	let fee = value / 2u32.into();
+	user: u32,
+	description: u32,
+) -> (
+	T::AccountId,
+	T::AccountId,
+	T::AssetKind,
+	BountyBalanceOf<T, I>,
+	BountyBalanceOf<T, I>,
+	T::Beneficiary,
+	Vec<u8>,
+) {
+	let caller = account("caller", user, SEED);
+	// Tiago: check with Muharem if we need coupling with pallet-assets
+	// let value: BalanceOf<T, I> = T::BountyValueMinimum::get().saturating_mul(100u32.into());
+	let asset_kind = <T as Config<I>>::BenchmarkHelper::create_asset_kind(SEED);
+	let value: BountyBalanceOf<T, I> = 100_000u32.into();
+	// Tiago: check with Muharem if we need coupling with pallet-assets
+	// TODO: revisit asset conversion
+	// let native_value =
+	// 	T::BalanceConverter::from_asset_balance(100u32.into(), asset_kind).unwrap_or(100u32.into());
+	let fee: BountyBalanceOf<T, I> = value / 2u32.into();
 	let deposit = T::BountyDepositBase::get() +
 		T::DataDepositPerByte::get() * T::MaximumReasonLength::get().into();
-	let _ = T::Currency::make_free_balance_be(&caller, deposit + minimum_balance::<T, I>());
-	let curator = account("curator", u, SEED);
-	let _ =
-		T::Currency::make_free_balance_be(&curator, fee / 2u32.into() + minimum_balance::<T, I>());
-	let reason = vec![0; d as usize];
-	(caller, curator, fee, value, reason)
+	let _ = T::Currency::make_free_balance_be(&caller, deposit + T::Currency::minimum_balance());
+	let curator = account("curator", user, SEED);
+	let curator_stash = account("curator_stash", user, SEED);
```

Comment:
you most probably also need the factory function for creating beneficiary value from some seed